File: binary search/ReverseString.java
public class ReverseString {

    public static char[] rev(char[] s, int l, int r) {
        if (l >= r) {
            return s;
        } else {
            char temp = s[l];
            s[l] = s[r];
            s[r] = temp;
            return rev(s, l + 1, r - 1);
        }
    }

    public static char[] reverseString(char[] s) {
        return rev(s, 0, s.length - 1);
    }

    public static void main(String[] args) {
        System.out.println(reverseString(new char[] { 'h', 'e', 'l', 'l', 'o' }));
        System.out.println(reverseString(new char[] { 'H', 'a', 'n', 'n', 'a', 'h' }));
    }
}

File: Coding-Upsolve/Reverse String.py
def reverseString(s):
    l, r = 0, len(s)-1

    while l < r:
        s[l], s[r] = s[r], s[l]
        l += 1
        r -= 1
    return s

print(reverseString(s = ["h","e","l","l","o"]))
# print(reverseString(s = ["H","a","n","n","a","h"]))